Modify rule S4156: Adapt to LaYC (#2678)
SonarSource/rspec: rules/S4156/javascript/rule.adoc

== Why is this an issue?

> Initially, TypeScript defined "internal modules" and "external modules".

Reviewer: ```suggestion
Initially, TypeScript defined "internal modules" and "external modules":
```

> * Internal modules: The `module` keyword was introduced in Typescript to define internal modules in Typescript. Internal modules were used to logically group classes, interfaces, functions into one unit.

Reviewer: ```suggestion
* Internal modules: The `module` keyword was introduced in TypeScript to define internal modules. Internal modules were used to group classes, interfaces, and functions into logical units.
```

* External modules refer to JavaScript modules, introduced in ECMAScript 2015. In TypeScript, just as in JavaScript (after ECMAScript 2015), any file containing a top-level `import` or `export` is considered a module.

> However, in order to avoid confusion with similarly named terms, `module` was deprecated in favor of the `namespace` keyword and "external modules" became simply "modules", as to align with ECMAScript 2015’s terminology.

Reviewer: ```suggestion
However, in order to avoid confusion with similarly named terms, `module` was deprecated in favor of the `namespace` keyword, and "external modules" became simply "modules", as to align with ECMAScript 2015’s terminology.
```
